File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include<string.h>
#include<math.h>

//count letters words and sentences
int count_letters(string s);
int count_words(string s);
int count_sentences(string s);
int main(void)
{
    string s = get_string("Text: ");
    float L = (float)count_letters(s) / (float)count_words(s) * 100;
    float S = (float)count_sentences(s) / (float)count_words(s) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n" , index);
    }
    return 0;
}

int count_letters(string s)
{
    int letters = 0;
    for (int i = 0 ; i < strlen(s) ; i++)
    {
        if ((s[i] >= 65 && s[i] <= 90) || (s[i] >= 97 && s[i] <= 122))
        {
            letters ++;
        }
    }
    return letters;
}

int count_words(string s)
{
    int words = 0;
    for (int i = 0 ; i < strlen(s) ; i++)
    {
        if(s[i] == 32)
        {
            words ++;
        }
    }
    return words + 1;
}

int count_sentences(string s)
{
    int sentences = 0;
    for (int i = 0 ; i < strlen(s) ; i++)
    {
        if (s[i] == '.' || s[i] == '!' || s[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

File: sentimental-readability/readability.py
from cs50 import get_string

text = get_string("Text: ")
letters = 0
words = 0
sentences = 0

for i in range(0, len(text)):
    if text[i].isalpha():
        letters += 1
    elif text[i] == " ":
        words += 1
    elif text[i] == "." or text[i] == "?" or text[i] == "!":
        sentences += 1
words += 1

L = letters / words * 100
S = sentences / words * 100
grade = round(0.0588 * L - 0.296 * S - 15.8)

if grade < 1:
    print("Before Grade 1")
if grade >= 16:
    print("Grade 16+")
else:
    print(f"Grade: {grade}")

